plugins/javaembed/javaembed.cpp

```diff
@@ -2874,10 +2888,27 @@ class JavaClassReader
             b.read(attr_count);
             for (unsigned j = 0; j < attr_count; j++)
             {
-                b.skip(2);
+                uint16_t attr_name_idx; b.read(attr_name_idx);
```

Comment:
This meant that you could not pass a dataset to an iterator<x> using embedded Java code.